### PR TITLE
Add FXIOS-6717 [v116] integrate redux in theming settings - Part 1

### PR DESCRIPTION
**Pull request by yoanarios**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6717)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15006)

### Description
Breaking down into two PRs before it gets more challenging to review, this first part includes:
- ThemeSettingsController clean up ex: Add missing dynamic text in the footer, remove an unused flag 
- Add Redux integration feature flag (unused ATM)
- Create a `MainState` app and reducer 
- Create `ThemeSettingsState`, `ThemeSettingsAction` and reducer

Next step: 
- Use feature flag
- Create a store and subscribe ThemeSettingsController to receive newState
- Dispatch actions to the store
- Analyze how to handle theme changes probably through Middleware so create ThemeMiddleware

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

File: nimbus-features/reduxIntegrationFeature.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+          Enables the feature
+        type: Boolean
+        default: false
+    defaults:
+      - channel: beta
+        value:
+          enabled: false
+      - channel: developer
+        value:
+          enabled: true
```
Comment (lmarceau) [inline nimbus-features/reduxIntegrationFeature.yaml:18]: Do we want this enabled `true` already?

**Review by OrlaM — APPROVED** [review]
One comment but it doesn't block this PR
LGTM
This is looking like it's going to be very easy to follow so really good work 👍

File: Client/Frontend/Settings/ThemeSettings/ThemeSettingsAction.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Redux
+
+enum ThemeSettingsAction: Action {
+    case enableSystemAppearance(Bool)
+    case toggleSwitchMode(SwitchMode)
+    case brightnessValueChanged(Float)
+    case updateUserBrightnessThreshold(Float)
```
[the remark this answers is not in the document]
Comment (yoanarios) [reply Client/Frontend/Settings/ThemeSettings/ThemeSettingsAction.swift:11]: Good point, I did miss this action 🙈 I will add it in the next PR

File: nimbus-features/reduxIntegrationFeature.yaml (hunk 1 above)
Comment (yoanarios) [reply nimbus-features/reduxIntegrationFeature.yaml:18]: I will disable in the next PR as there is no blocking issue for this one

**Review by lmarceau — APPROVED** [review]

**Pull request merged**